core/libp2p/peer/protocol_repository/inmem_protocol_repository.hpp

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_INMEM_PROTOCOL_REPOSITORY_HPP
+#define KAGOME_INMEM_PROTOCOL_REPOSITORY_HPP
+
+#include <set>
+#include <unordered_map>
+
+#include "libp2p/peer/protocol_repository.hpp"
+
+namespace libp2p::peer {
+
+  class InmemProtocolRepository : public ProtocolRepository {
```

Comment:
I think it would be better to leave a small comment here, explaining how the class implements ProtocolRepository in a nutshell.